pkg/informers/informer.go

```diff
@@ -293,3 +304,27 @@ func (i *multiNamespaceInformer) SetTransform(handler cache.TransformFunc) error
 	}
 	return errors.NewAggregate(errList)
 }
+
+func (i *multiNamespaceInformer) RemoveEventHandler(handle cache.ResourceEventHandlerRegistration) error {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	for _, inf := range i.informers {
+		if err := inf.RemoveEventHandler(handle); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (i *multiNamespaceInformer) IsStopped() bool {
```

Comment:
You probably need to add the following guard clause at the start of this function:

```suggestion
func (i *multiNamespaceInformer) IsStopped() bool {
	if !i.namespaces.Initialized() {
		return true   // or false?
	}
```

There's a similar check in `HasSynced()`. We had to add it, because the function was called *before* the namespaces were initialized. This caused the function to return `true` when it wasn't yet synced. A similar thing might happen here in `IsStopped`, but I don't know what exactly "stopped" means: before the namespaces are initialized, should the informer be considered stopped or not?

Reply:
Thanks!

> but I don't know what exactly "stopped" means: before the namespaces are initialized, should the informer be considered stopped or not?

Good question. I checked how sharedIndexInformer is implemented and it works as follows:
```
// IsStopped reports whether the informer has already been stopped
func (s *sharedIndexInformer) IsStopped() bool {
	s.startedLock.Lock()
	defer s.startedLock.Unlock()
	return s.stopped
}
```
and `stopped` varaible is `true` only when the informer was running and then was stopped.
```
func (s *sharedIndexInformer) Run(stopCh <-chan struct{}) {
        ...
        defer func() {
		s.startedLock.Lock()
		defer s.startedLock.Unlock()
		s.stopped = true // Don't want any new listeners
	}()
	s.controller.Run(stopCh)
}
```
so I think we should return `false` if namespace set was not initialized.